astrokat/noisediode.py

```diff
@@ -9,257 +9,408 @@
     from katcorelib import user_logger
 except ImportError:
     from .simulate import user_logger
+from . import _DEFAULT_LEAD_TIME
+from . import max_cycle_len
 
 
-_DEFAULT_LEAD_TIME = 5.0  # lead time [sec]
+def _get_max_cycle_len(kat):
+    """Get maximum cycle length for noise diode switching
+    """
+    if not kat.dry_run:
+        return max_cycle_len(kat.sensor.sub_band.get_value())
+    else:
+        return max_cycle_len('l')
 
 
-def _katcp_reply_to_log_(dig_katcp_replies):
-    for ant in sorted(dig_katcp_replies):
-        reply, informs = dig_katcp_replies[ant]
-        # assuming ND for all antennas set at the same time
-        # only return timestamp of the last antenna set
-        timestamp = _nd_log_msg_(ant, reply, informs)
-    return timestamp
+def _get_nd_timestamp_(lead_time):
+    """Timestamp for ND switch command with lead time
+    """
+    return time.time() + lead_time
+
+
+def _set_dig_nd_(kat,
+                 timestamp,
+                 nd_setup=None,  # no pattern set
+                 switch=0,
+                 cycle=False):
+    """Setting and implementing digitiser noise diode command
+    Parameters
+    ----------
+    kat : session kat container-like object
+        Container for accessing KATCP resources allocated to schedule block.
+    timestamp : float
+        Linux timestamp in seconds at which to switch noise diode
+    nd_setup : dict, optional (default = None)
+        Noise diode pattern setup, with keys:
+            'antennas':  options are 'all', or 'm062', or ....,
+            'cycle_len': the cycle length [sec],
+                           - must be less than 20 sec for L-band,
+            etc., etc.
+    switch : 0 or 1, optional (default = 0)
+        Switch all noise diodes off (0) or on (1)
+    """
+
+    if nd_setup is not None:
+        # selected antennas for nd pattern
+        nd_antennas = nd_setup['antennas'].split(",")
+        # nd pattern length [sec]
+        cycle_length = nd_setup['cycle_len']
+        # on fraction of pattern length [%]
+        on_fraction = nd_setup['on_frac']
+        msg = ('Repeat noise diode pattern every {} sec, '
+               'with {} sec on and apply pattern to {}'
+               .format(cycle_length,
+                       float(cycle_length) * float(on_fraction),
+                       nd_antennas))
+        user_logger.info(msg)
+    else:
+        nd_antennas = [ant.name for ant in kat.ants]
```

Comment:
Perhaps sort the ant names here instead.
```suggestion
        nd_antennas = sorted(ant.name for ant in kat.ants)
```

Reply:
added sorted option at assignment as suggested